docs/eventing/sources/pingsource.md

```diff
@@ -1,196 +1,257 @@
+        - image: gcr.io/knative-releases/knative.dev/eventing-contrib/cmd/event-display
+EOF
+```
+{{< /tab >}}
 
----
+{{% tab name="kn" %}}
+Create a service using the `kn` CLI:
 
-kind: Service
-apiVersion: v1
-metadata:
-  name: event-display
-spec:
-  selector:
-    app: event-display
-  ports:
-  - protocol: TCP
-    port: 80
-    targetPort: 8080
-EOF
+```shell
+kn service create event-display --image gcr.io/knative-releases/knative.dev/eventing-contrib/cmd/event-display
 ```
+{{< /tab >}}
+{{< /tabs >}}
+
+## Create a producer
 
-### Creating the PingSource
+Create a producer that sends events to your consumer every two minutes.
+The `sink` element in the metadata describes where to send events.
+In this case, events are sent to a service with the name `event-display`.
 
-You can now create the `PingSource` sending an event containing
-`{"message": "Hello world!"}` every minute.
+For each set of ping events that you want to request, create an event source in the same namespace
+as the destination.
 
 {{< tabs name="create-source" default="YAML" >}}
 {{% tab name="YAML" %}}
+Create the event source from standard in by running:
 
 ```shell
-kubectl create -n pingsource-example -f - <<EOF
-apiVersion: sources.knative.dev/v1beta2
+cat <<EOF | kubectl create -f -
+apiVersion: sources.knative.dev/v1alpha2
 kind: PingSource
 metadata:
   name: test-ping-source
 spec:
-  schedule: "*/1 * * * *"
+  schedule: "*/2 * * * *"
   contentType: "application/json"
   data: '{"message": "Hello world!"}'
   sink:
     ref:
-      apiVersion: v1
+      apiVersion: serving.knative.dev/v1
       kind: Service
       name: event-display
 EOF
 ```
-
 {{< /tab >}}
 
 {{% tab name="kn" %}}
-Notice that the namespace is specified in two places in the command in `--namespace` and the `--sink` hostname
+Create a PingSource by using the `kn` CLI:
+
 ```shell
 kn source ping create test-ping-source \
-  --namespace pingsource-example \
-  --schedule "*/1 * * * *" \
-  --data '{"message": "Hello world!"}' \
-  --sink http://event-display.pingsource-example.svc.cluster.local
+  --schedule "*/2 * * * *" --data \
+  '{ "message": "Hello world!" }' \
+  --sink ksvc:event-display
 ```
-
 {{< /tab >}}
-{{< /tabs >}}
 
-## (Optional) Create a PingSource with binary data
+{{% tab name="binary" %}}
+You can send binary data, that cannot be directly serialized by using YAML, to downstream, by using
+`dataBase64` as the payload. `dataBase64` carries data that is base64-encoded.
 
-Sometimes you may want to send binary data, which cannot be directly serialized in yaml, to downstream. This can be achieved by using `dataBase64` as the payload. As the name suggests, `dataBase64` should carry data that is base64 encoded.
+**Note:** `data` and `dataBase64` cannot co-exist.
 
-Please note that `data` and `dataBase64` cannot co-exist.
+Create the event source with binary data from standard in by running:
```

Comment:
I'd leave out "from standard" entirely here.
As an example, I'd prefer
"To create an event source by using binary data, enter the command:"

Reply:
Alternatively, I might call this "stdin", "standard input" (weird, but what STDIN is abbreviated from) or "terminal input".